core/invoker/src/main/scala/whisk/core/containerpool/ContainerPool.scala

```diff
@@ -71,6 +71,8 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
   var freePool = immutable.Map.empty[ActorRef, ContainerData]
   var busyPool = immutable.Map.empty[ActorRef, ContainerData]
   var prewarmedPool = immutable.Map.empty[ActorRef, ContainerData]
+  var lastRetryLog: Option[Long] = None
```

Comment:
I don't think we want to globally throttle, but throttle per message. The current code would different messages being swallowed.

That's actually why @vvraskin created the `retries` field of `Run`. In the same sense I think you'd want to create a `lastRetry: Instant` field, which can be `Instant.EPOCH` by default.

Reply:
Though we saw that only one message is looping at a given time, I agree that it is better to track the time for each run seperately. Changed